### PR TITLE
[xbar/dv] xbar TB automation update

### PR DESCRIPTION
**Pull request by weicaiyang**
1. auto-gen bind.core
2. make TB not depend on any specific top

Signed-off-by: Weicai Yang <weicai@google.com>

### CODE REVIEW

File: util/topgen.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ def generate_xbars(top, out_path):
         xbar_path = out_path / 'ip/xbar_{}/data/autogen'.format(obj["name"])
         xbar_path.mkdir(parents=True, exist_ok=True)
         xbar = tlgen.validate(obj)
+        xbar.top_name = "top_" + top["name"]
```
Comment (sriyerg) [inline util/topgen.py]: +@eunchan

File: util/tlgen/xbar.sim_cfg.hjson.tpl [hunk 2]
```diff
@@ -7,6 +7,12 @@
   // Top level dut name (sv module).
   dut: xbar_${xbar.name}
 
+  // Testplan hjson file.
+  testplan: "{proj_root}/hw/${xbar.top_name}/ip/{dut}/data/autogen/{dut}_testplan.hjson"
+
+  // Add xbar specific exclusion files.
+  vcs_cov_excl_files: ["{proj_root}/hw/${xbar.top_name}/ip/{dut}/dv/cov/xbar_cov_excl.el"]
+
```
Comment (eunchan) [inline util/tlgen/xbar.sim_cfg.hjson.tpl]: If anything is related to the top, then I recommend it to be generated in topgen. Same to `top_name` field. The value is assigned in the topgen, but revised in the tlgen data structure.

File: util/tlgen/xbar.py [hunk 3]
```diff
@@ -22,6 +22,7 @@ class Xbar:
     name = ""  # str  # e.g. "main" --> main_xbar
     clocks = []  # All clocks of xbar
     resets = []  # All resets of xbar
+    top_name = ""  # name of chip
```
Comment (eunchan) [inline util/tlgen/xbar.py]:
```suggestion
    ip_path = ""  # The Xbar RTL/DV path
```

File: util/tlgen/xbar.sim_cfg.hjson.tpl [hunk 4]
```diff
@@ -7,6 +7,12 @@
   // Top level dut name (sv module).
   dut: xbar_${xbar.name}
 
+  // Testplan hjson file.
+  testplan: "{proj_root}/hw/${xbar.top_name}/ip/{dut}/data/autogen/{dut}_testplan.hjson"
```
Comment (eunchan) [inline util/tlgen/xbar.sim_cfg.hjson.tpl]:
```suggestion
  testplan: "{proj_root}/${xbar.ip_path}/{dut}/data/autogen/{dut}_testplan.hjson"
```

File: util/tlgen/xbar.sim_cfg.hjson.tpl [hunk 5]
```diff
@@ -7,6 +7,12 @@
...
+
+  // Add xbar specific exclusion files.
+  vcs_cov_excl_files: ["{proj_root}/hw/${xbar.top_name}/ip/{dut}/dv/cov/xbar_cov_excl.el"]
```
Comment (eunchan) [inline util/tlgen/xbar.sim_cfg.hjson.tpl]: ditto

File: util/topgen.py (hunk 1 above)
Comment (eunchan) [inline util/topgen.py]:
```suggestion
        xbar.ip_path = "hw/top_" + top["name"] + "/ip"
```

Comment (weicaiyang) [reply util/topgen.py]: Thanks @eunchan for the suggestion.
As discussed, change it to ip_path and make sure tlgen.py can generate xbar directly. PTAL

**Pull request merged**